sdk/synapse/azure-synapse-accesscontrol/azure/synapse/accesscontrol/aio/operations/_role_assignments_operations.py

```diff
@@ -40,128 +50,101 @@ def __init__(self, client, config, serializer, deserializer) -> None:
         :return: CheckPrincipalAccessResponse, or the result of cls(response)
         :rtype: ~azure.synapse.accesscontrol.models.CheckPrincipalAccessResponse
         :raises: ~azure.core.exceptions.HttpResponseError
         """
-        cls = kwargs.pop('cls', None)  # type: ClsType["_models.CheckPrincipalAccessResponse"]
-        error_map = {
-            401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError
-        }
-        error_map.update(kwargs.pop('error_map', {}))
+        cls = kwargs.pop("cls", None)  # type: ClsType["_models.CheckPrincipalAccessResponse"]
+        error_map = {401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError}
+        error_map.update(kwargs.pop("error_map", {}))
+
+        content_type = kwargs.pop("content_type", "application/json")  # type: Optional[str]
 
         _request = _models.CheckPrincipalAccessRequest(subject=subject, actions=actions, scope=scope)
-        api_version = "2020-08-01-preview"
-        content_type = kwargs.pop("content_type", "application/json")
-        accept = "application/json, text/json"
+        json = self._serialize.body(_request, "object")
 
-        # Construct URL
-        url = self.check_principal_access.metadata['url']  # type: ignore
+        request = rest_role_assignments.build_check_principal_access_request(
+            json=json, content_type=content_type, template_url=self.check_principal_access.metadata["url"], **kwargs
+        )._internal_request
         path_format_arguments = {
-            'endpoint': self._serialize.url("self._config.endpoint", self._config.endpoint, 'str', skip_quote=True),
+            "endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, "str", skip_quote=True),
         }
-        url = self._client.format_url(url, **path_format_arguments)
-
-        # Construct parameters
-        query_parameters = {}  # type: Dict[str, Any]
-        query_parameters['api-version'] = self._serialize.query("api_version", api_version, 'str')
-
-        # Construct headers
-        header_parameters = {}  # type: Dict[str, Any]
-        header_parameters['Content-Type'] = self._serialize.header("content_type", content_type, 'str')
-        header_parameters['Accept'] = self._serialize.header("accept", accept, 'str')
-
-        body_content_kwargs = {}  # type: Dict[str, Any]
-        body_content = self._serialize.body(_request, 'CheckPrincipalAccessRequest')
-        body_content_kwargs['content'] = body_content
-        request = self._client.post(url, query_parameters, header_parameters, **body_content_kwargs)
-        pipeline_response = await self._client._pipeline.run(request, stream=False, **kwargs)
+        request.url = self._client.format_url(request.url, **path_format_arguments)
+
+        pipeline_response = await self._client.send_request(
+            request, stream=False, _return_pipeline_response=True, **kwargs
```

Comment:
what is _return_pipeline_response?

Reply:
ooh good point. In my new version of azure.core.rest, I'm using that kwarg, but not in this older version of `azure.core.rest`. After the bugbash today, I'll change this part of the code so that it can still run. The code will eventually look like this, once we have in `azure.core.rest`